### PR TITLE
Initial support for Workspaces and Panels (split and join)

### PR DESCRIPTION
**Pull request by erri120**
Part of #674.

- Adds a design-only playground for testing. This is currently the only place where the system is used, so the feature is completely isolated. The code for the playground is in constant change, so don't read into it too much.
- Adds a `Workspace` and `Panel` component:
  - Panels can be added (split) and removed (join).
  - Math is outside the VM and fully tested.

### CODE REVIEW

File: src/NexusMods.App.UI/WorkspaceSystem/Panel/IPanelViewModel.cs [hunk 1]
```diff
@@ -0,0 +1,35 @@
+using Avalonia;
+
+namespace NexusMods.App.UI.WorkspaceSystem;
+
+public interface IPanelViewModel : IViewModelInterface
```
Comment (Al12rs) [inline src/NexusMods.App.UI/WorkspaceSystem/Panel/IPanelViewModel.cs:5]: I would add a comment describing what a panel is in this context and its purpose.

Comment (erri120) [reply src/NexusMods.App.UI/WorkspaceSystem/Panel/IPanelViewModel.cs:5]: I'll add documentation once I add tabs to the system.

File: src/NexusMods.App.UI/WorkspaceSystem/Workspace/IWorkspaceViewModel.cs [hunk 2]
```diff
@@ -0,0 +1,26 @@
+using System.Collections.ObjectModel;
+using System.Reactive;
+using Avalonia;
+using ReactiveUI;
+
+namespace NexusMods.App.UI.WorkspaceSystem;
+
+public interface IWorkspaceViewModel : IViewModelInterface
```
Comment (Al12rs) [inline src/NexusMods.App.UI/WorkspaceSystem/Workspace/IWorkspaceViewModel.cs:8]: Here as well, a comment describing what a Workspace is and its purpose. We had some naming issues on these so having some clarification in the code is important I think.

Comment (erri120) [reply src/NexusMods.App.UI/WorkspaceSystem/Workspace/IWorkspaceViewModel.cs:8]: I'll add documentation once I add tabs to the system.

**Review by Al12rs — APPROVED** [review]

**Pull request merged**